Index: hfsprogs-332.14/fsck_hfs.tproj/dfalib/SControl.c
===================================================================
--- hfsprogs-332.14.orig/fsck_hfs.tproj/dfalib/SControl.c	2008-10-07 05:08:02.000000000 -0300
+++ hfsprogs-332.14/fsck_hfs.tproj/dfalib/SControl.c	2008-10-07 05:08:14.000000000 -0300
@@ -739,7 +739,7 @@
 		pointer = (ScavStaticStructures *) AllocateClearMemory( sizeof(ScavStaticStructures) );
 		if ( pointer == nil ) {
 			if ( GPtr->logLevel >= kDebugLog ) {
-				printf( "\t error %d - could not allocate %ld bytes of memory \n",
+				printf( "\t error %d - could not allocate %i bytes of memory \n",
 					R_NoMem, sizeof(ScavStaticStructures) );
 			}
 			return( R_NoMem );
Index: hfsprogs-332.14/fsck_hfs.tproj/dfalib/hfs_endian.c
===================================================================
--- hfsprogs-332.14.orig/fsck_hfs.tproj/dfalib/hfs_endian.c	2008-10-07 05:08:08.000000000 -0300
+++ hfsprogs-332.14/fsck_hfs.tproj/dfalib/hfs_endian.c	2008-10-07 05:08:14.000000000 -0300
@@ -563,7 +563,7 @@
             /* Make sure name length is consistent with key length */
             if (keyLength < sizeof(srcKey->parentID) + sizeof(srcKey->nodeName.length) +
                 srcKey->nodeName.length*sizeof(srcKey->nodeName.unicode[0])) {
-				if (debug) printf("hfs_swap_HFSPlusBTInternalNode: catalog record #%d keyLength=%d expected=%lu\n",
+				if (debug) printf("hfs_swap_HFSPlusBTInternalNode: catalog record #%d keyLength=%d expected=%i\n",
 					srcDesc->numRecords-i, keyLength, sizeof(srcKey->parentID) + sizeof(srcKey->nodeName.length) +
                     srcKey->nodeName.length*sizeof(srcKey->nodeName.unicode[0]));
 				WriteError(fcb->fcbVolume->vcbGPtr, E_KeyLen, fcb->fcbFileID, src->blockNum);
