--- a/fsck_hfs.tproj/dfalib/SVerify1.c
+++ b/fsck_hfs.tproj/dfalib/SVerify1.c
@@ -2176,7 +2176,7 @@
 	/* Always update the last recordType, fileID and attribute name before exiting */
 	prevAttr->recordType = rec->recordType;
 	prevAttr->fileID = fileID;
-	(void) strlcpy((char *)prevAttr->attrname, (char *)attrname, sizeof(prevAttr->attrname));
+	(void) strncpy((char *)prevAttr->attrname, (char *)attrname, sizeof(prevAttr->attrname));
 
 	goto out;
 
@@ -3201,7 +3201,7 @@
 		if (extentInfo.attrname == NULL) {
 			return(memFullErr);
 		}
-		strlcpy(extentInfo.attrname, attrname, attrlen);
+		strncpy(extentInfo.attrname, attrname, attrlen);
 	}
 	
 	//	If it's uninitialized
