Index: hfsprogs-332.14/newfs_hfs.tproj/makehfs.c
===================================================================
--- hfsprogs-332.14.orig/newfs_hfs.tproj/makehfs.c	2007-07-19 23:40:11.000000000 -0300
+++ hfsprogs-332.14/newfs_hfs.tproj/makehfs.c	2007-07-19 23:40:11.000000000 -0300
@@ -70,7 +70,7 @@
 #include "readme.h"
 
 
-#define HFS_BOOT_DATA	"/usr/share/misc/hfsbootdata"
+#define HFS_BOOT_DATA	"/usr/share/hfsprogs/hfsbootdata"
 
 #define HFS_JOURNAL_FILE	".journal"
 #define HFS_JOURNAL_INFO	".journal_info_block"
